src/main/webapp/app/markdown-editor/markdown-editor.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, AfterViewInit, ViewChild, Input, Output, EventEmitter } from '@angular/core';
+import { AceEditorComponent } from 'ng2-ace-editor';
+import 'brace/theme/chrome';
+import 'brace/mode/markdown';
+import { Command } from 'app/markdown-editor/commands/command';
+import { BoldCommand } from 'app/markdown-editor/commands/bold.command';
+import { ItalicCommand } from 'app/markdown-editor/commands/italic.command';
+import { UnderlineCommand } from 'app/markdown-editor/commands/underline.command';
+import { ArtemisMarkdown } from 'app/components/util/markdown.service';
+import { HintCommand } from 'app/markdown-editor/specialcommands/hint.command';
+import { CorrectoptionCommand } from 'app/markdown-editor/specialcommands/correctoption.command';
+import { IncorrectoptionCommand } from 'app/markdown-editor/specialcommands/incorrectoption.command';
+import { ExplanationCommand } from 'app/markdown-editor/specialcommands/explanation.command';
+
+@Component({
+    selector: 'jhi-markdown-editor',
+    styleUrls: ['./markdown-editor.scss'],
+    providers: [ArtemisMarkdown],
+    templateUrl: './markdown-editor.component.html'
+})
+export class MarkdownEditorComponent implements AfterViewInit {
+    @ViewChild('aceEditor')
+    aceEditorContainer: AceEditorComponent;
+
+    @Input() defaultText: string;
+
+    @Output() defaultTextChanged = new EventEmitter();
+
+    questionEditorText = '';
+    questionEditorAutoUpdate = true;
+
+    hintCommand = new HintCommand();
+    correctCommand = new CorrectoptionCommand();
+    incorrectCommand = new IncorrectoptionCommand();
+    explanationCommand = new ExplanationCommand();
```

Comment:
I think those Commands should be injected at runtime to keep the Editor Flexible. 
There should be an `@Input() specialCommands: SpecialCommand[]` and those need to be merged into the `commands` Array.
This required the adapted Inheritance Structure ( `Command` ◁-- `SpecialCommand` ) to work in a sane fashion.